Source/JavaScriptCore/runtime/JSCJSValueInlines.h

```diff
@@ -1069,8 +1069,11 @@ ALWAYS_INLINE bool JSValue::getPropertySlot(JSGlobalObject* globalObject, Proper
         }
         object = synthesizePrototype(globalObject);
         EXCEPTION_ASSERT(!!scope.exception() == !object);
-        if (UNLIKELY(!object))
+        if (UNLIKELY(!object)) {
+            VM& vm = getVM(globalObject);
+            RELEASE_ASSERT(vm.exceptionForInspection() && vm.traps().maybeNeedHandling(), vm.traps().maybeNeedHandling(), vm.exceptionForInspection(), JSValue::encode(*this));
             return false;
```

Comment:
This is one of the very hot path. Let's not add a cost to that, and make it scoped to AbstractModuleRecord.cpp.